lib/Conversion/SMTToZ3LLVM/LowerSMTToZ3LLVM.cpp

```diff
@@ -822,6 +1033,12 @@ void circt::populateSMTToZ3LLVMConversionPatterns(
                                        converter, patterns.getContext(),       \
                                        globals, options, APINAME,              \
                                        MIN_NUM_ARGS);
+
+#define ADD_ONE_TO_ONE_PATTERN(OP, APINAME, NUM_ARGS)                          \
+  patterns.add<OneToOneSMTPattern<OP>>(/*NOLINT(bugprone-macro-parentheses)*/  \
+                                       converter, patterns.getContext(),       \
+                                       globals, options, APINAME, NUM_ARGS);
+
```

Comment:
Same comment as in #6902 about potentially making these templated lambdas to avoid a macro.

Reply:
AFAIK templated lambdas are a C++20 feature. If there is something similar we can use in C++17, let me know!